app/src/main/java/com/goldenraven/devkitwallet/data/ElectrumServer.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021 thunderbiscuit and contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the ./LICENSE file.
+ */
+
+package com.goldenraven.devkitwallet.data
+
+import android.util.Log
+import com.goldenraven.devkitwallet.utilities.TAG
+import org.bitcoindevkit.Blockchain
+import org.bitcoindevkit.BlockchainConfig
+import org.bitcoindevkit.ElectrumConfig
+
+class ElectrumServer {
+    private var useDefaultElectrum: Boolean = true
+    private var default: Blockchain
+    private var custom: Blockchain? = null
+    private var electrumURL: String
+
+    init {
+        electrumURL = "ssl://electrum.blockstream.info:60002"
+        val blockchainConfig = BlockchainConfig.Electrum(ElectrumConfig(electrumURL, null, 5u, null, 10u))
+        default = Blockchain(blockchainConfig)
+    }
+
+    val server: Blockchain =  if (useDefaultElectrum) this.default else this.custom!!
```

Comment:
This form will not produce what you intend it to: the if/else will only be triggered once, upon creation of the `server` variable which will be set to either of `default` or `custom`, but will then be that forevermore.

Instead you'll need to use [custom getter syntax](https://kotlinlang.org/docs/properties.html#getters-and-setters), which will make the `server` value return the proper Blockchain object every time it is accessed.

```kotlin
val server: Blockchain
    get() = if (useDefaultElectrum) this.default else this.custom!!
```
```